src/components/Textarea/index.tsx

```diff
@@ -0,0 +1,102 @@
+import type { CSSProperties, HTMLAttributes, ReactElement } from 'react'
+import { useCallback, useEffect, useRef } from 'react'
+import styled from '@emotion/styled'
+
+export interface TextAreaProps extends HTMLAttributes<HTMLDivElement> {
+  label?: string
+  placeholder?: string
+  children: string
+  guideMessage?: string
+  textAreaStyle?: CSSProperties
+  BgType: 'filled' | 'ghost'
+  autoFocus?: boolean
+}
+
+export const TextArea = ({
+  label,
+  placeholder = '내용을 입력하세요.',
+  children,
+  guideMessage,
+  BgType,
+  textAreaStyle,
+  autoFocus,
+  ...props
+}: TextAreaProps): ReactElement => {
+  const ref = useRef<HTMLTextAreaElement>(null)
+  const isRefValueNull = ref === null || ref.current === null
+  const handleResizeHeight = useCallback(() => {
+    if (isRefValueNull) {
+      return
+    }
+    ref.current.style.height = '120px'
+    ref.current.style.height = ref.current.scrollHeight + 'px'
+  }, [])
+
+  useEffect(() => {
+    if (ref === null || ref.current === null) {
+      return
+    }
+    ref.current.style.height = '120px'
+    ref.current.style.height = ref.current.scrollHeight + 'px'
+  }, [])
```

Comment:
로직이 중복되고 있어서 이미 정의된 함수를 사용하는건 어떨까용 ?
```suggestion
  const handleResizeHeight = useCallback(() => {
    if (isRefValueNull) {
      return
    }
    ref.current.style.height = '120px'
    ref.current.style.height = ref.current.scrollHeight + 'px'
  }, [])

  useEffect(() => {
   handleResizeHeight()
  }, [])
```